src/dsp_tools/utils/connection.py

```diff
@@ -15,18 +15,21 @@ class Connection(Protocol):
     def get(
         self,
         route: str,
+        headers: Optional[dict[str, str]] | None = None,
     ) -> dict[str, Any]:
         """
         Make a HTTP GET request to the server to which this connection has been established.
 
         Args:
             route: route that will be called on the server
+            headers: headers for the HTTP request
         """
 
     def put(
         self,
         route: str,
         jsondata: str | None = None,
+        headers: Optional[dict[str, str]] | None = None,
```

Comment:
No need for optional when you define it as union with None (Optional is a relict from a time befor python supported nice union syntax)
```suggestion
        headers: dict[str, str] | None = None,
```